File: code/python/chainofresponsibility/LevelOneSupport.py
from SupportLevel import SupportLevel

class LevelOneSupport(SupportLevel):
    def __init__(self):
        self.next = None

    def set_next(self, next):
        print(f"Level one support setting next level to {type(next).__name__} Support")
        self.next = next

    def handle_request(self, ticket):
        if ticket.get_level() <= 1:
            print(f"Level One Support handling ticket: {ticket.get_description()}")
            print("Ticket resolved.")
        else:
            print(f"Level One Support cannot handle ticket: {ticket.get_description()} Escalating to next level.")
            self.next.handle_request(ticket)
            print("Ticket returned to level one after being resolved.")

File: code/java/src/main/java/chainofresponsibility/LevelOneSupport.java
package chainofresponsibility;

class LevelOneSupport implements SupportLevel {
    private SupportLevel next;

    @Override
    public void setNext(SupportLevel next) {

        System.out.println("Level one support setting next level to " + next.getClass().getSimpleName() + " Support");
        this.next = next;
    }

    @Override
    public void handleRequest(SupportTicket ticket) {
        if (ticket.getLevel() <= 1) {
            System.out.println("Level One Support handling ticket: " + ticket.getDescription());
            System.out.println("Ticket resolved.");
        } else {
            System.out.println("Level One Support cannot handle ticket: " + ticket.getDescription() + " Escalating to next level.");
            next.handleRequest(ticket);
            System.out.println("Ticket returned to level one after being resolved.");
        }
    }
}
